src/braket/aws/aws_quantum_task.py

```diff
@@ -255,6 +256,10 @@ def _status(self, use_cached_value=False):
             self._logger.warning(f"Task is in terminal state {status} and no result is available")
         return status
 
+    def _update_status_if_nonterminal(self):
```

Comment:
This is still not ideal: consider the case where the metadata is empty and the status is "RUNNING".
Then first self._status(True) is called, filling the metadata and returning the "RUNNING" status.
Since it's not a terminal state, self._status(False) will be called again. This uses two GetQuantumTask calls.

Reply:
However, there will be at most one extra call to GetQuantumTask, and since the status isn't terminal, I don't think it hurts to make the extra call; it's going to have to poll again regardless. Easy enough to get rid of it though.

Reply:
In the scenario I described it's two successive calls to GetQuantumTask, so the second call will likely return the same result and be wasted. In any case, thanks for the fix.